File: backend/app/models/srp_destination.rb
# == Schema Information
#
# Table name: srp_destinations
#
#  id             :integer          not null, primary key
#  name           :integer          not null
#  description    :string(255)
#  corporation_id :integer
#  alliance_id    :integer
#  external       :string(255)
#  created_at     :datetime         not null
#  updated_at     :datetime         not null
#

class SrpDestination < ActiveRecord::Base
  include Ng2SearchTableSearchable

  # Constants
  RANSACK_FILTER_ATTRIBUTES = {
    id: :id_eq,
    name: :name_cont,
    description: :description_cont
  }.with_indifferent_access.freeze

  RANSACK_SORT_ATTRIBUTES = {
    id: :id,
    name: :name,
    created_at: :created_at,
    updated_at: :updated_at,
  }.with_indifferent_access.freeze

  # Relations
  belongs_to :corporation

  # Validations
  validates :name, presence: true, length: { maximum: 255 }
  validates :description, length: { maximum: 255 }

  # Hooks

  # Scopes

  # Methods


end

File: frontend/src/app/models/srp-destination.ts
import {JsonProperty} from '../common/json-property/json-property';

export class SrpDestination {
  @JsonProperty()
  id: number;

  @JsonProperty()
  name: string;

  @JsonProperty('corporation_id')
  corporationId: number;

  @JsonProperty('alliance_id')
  allianceId: number;

  @JsonProperty()
  external: string;

  @JsonProperty()
  description: string;

  @JsonProperty('created_at')
  createdAt: string;

  @JsonProperty('updated_at')
  updatedAt: string;

  constructor(obj?: any) {
    this.id = obj && obj.id;
    this.name = obj && obj.name;
    this.corporationId = obj && obj.corporationId;
    this.allianceId = obj && obj.allianceId;
    this.external = obj && obj.external;
    this.description = obj && obj.description;
    this.createdAt = obj && obj.createdAt;
    this.updatedAt = obj && obj.updatedAt;
  }
}
